File: assign3/src/vehicles/Vehicle.java
package vehicles;

public class Vehicle{

  private int _capacity;
  private int _occupants;
  private int _roomLeft;

  public int getCapacity() { return _capacity; }
  public void setCapacity(int capacity){
    _capacity = capacity;
  }

  private void _calcRoomLeft(){
    _roomLeft = _capacity - _occupants;
  }

  public int getRoomLeft(){
    _calcRoomLeft();
    return _roomLeft;
  }

  public int getOccupants(){ return _occupants; }
  public void setOccupants(int occupants) {
    _occupants = occupants;
    _calcRoomLeft();
  }

}

File: Assignment5/src/Vehicle.cpp
#include "Vehicle.h"

namespace vehicles{
  Vehicle::Vehicle(){}

  int Vehicle::getCapacity() const{
    return _capacity;
  }
  void Vehicle::setCapacity(int capacity){
    _capacity = capacity;
  }
  void Vehicle::_calcRoomLeft(){
    _roomLeft = _capacity - _occupants;
  }
  int Vehicle::getRoomLeft(){
    _calcRoomLeft();
    return _roomLeft;
  }
  int Vehicle::getOccupants() const{
    return _occupants;
  }
  void Vehicle::setOccupants(int occupants){
    _occupants = occupants;
    _calcRoomLeft();
  }
  Vehicle::~Vehicle(){}
}
